autoplex/data/common/jobs.py

```diff
@@ -648,24 +685,65 @@ def safe_strip_hostname(value):
 
 
 @job
-def Data_preprocessing(
+def preprocess_data(
     vasp_ref_dir: str,
-    test_ratio: float = 0.5,
+    test_ratio: float | None = None,
     regularization: bool = False,
+    scheme: str = "linear-hull",
     distillation: bool = False,
-    f_max: float = 40.0,
+    force_max: float = 40,
     force_label: str = "REF_forces",
-    pre_database_dir: str = None,
-    etup: list[tuple] = None,
-):
-    """Preprocesse data to a suitable format for fiting machine learning models."""
+    pre_database_dir: str | None = None,
+    reg_minmax: list[tuple] | None = None,
+    isolated_atom_energies: dict | None = None,
+) -> Path:
+    """
+    Preprocesse data to before fiting machine learning models.
+
+    This function handles tasks such as splitting the dataset,
+    applying regularization, accumulating database, and filtering
+    structures based on maximum force values.
+
+    Parameters
+    ----------
+    vasp_ref_dir: str
+        Path to the directory containing the reference VASP calculation data.
+    test_ratio: float
+        The proportion of the test set after splitting the data.
+        If None, no splitting will be performed.
+    regularization: bool
+        If true, apply regularization. This only works for GAP.
+    scheme: str
+        Scheme to use for regularization.
+    distillation: bool
+        If True, apply data distillation.
+    force_max: float
+        Maximum force value to exclude structures.
+    force_label: str
+        The label of force values to use for distillation.
+    pre_database_dir : str
+        Directory where the previous database was saved.
+    reg_minmax: list[tuple]
+        A list of tuples representing the minimum and maximum
+        values for regularization.
+    isolated_atom_energies: dict
+        A dictionary containing isolated energy values for different species.
+
+    Returns
+    -------
+    Path
+        The current working directory.
+    """
     atoms = (
-        data_distillation(vasp_ref_dir, f_max, force_label)
+        data_distillation(vasp_ref_dir, force_max, force_label)
         if distillation
         else read(vasp_ref_dir, index=":")
     )
 
-    train_structures, test_structures = stratified_dataset_split(atoms, test_ratio)
+    if test_ratio == 0 or test_ratio is None:
+        train_structures, test_structures = atoms, atoms
```

Comment:
Okay, great. This solves one of my issues as well. Thus, we train, test on the same data if not enough data points are there for a stratified split.